### DISCUSSION TITLE
NOTICE: New (~5/30?) NVidia drivers change memory management

### DISCUSSION THREAD

**Discussion by cgessai** [LostRuins/koboldcpp#262] [General]
PREVIOUS BEHAVIOR: If more VRAM was requested than was available, error would result

NEW BEHAVIOR: If more VRAM is requested than available, drivers seamlessly and automatically dip into Shared GPU Memory (i.e. RAM).

[More information](https://github.com/vladmandic/automatic/discussions/1285), but related to Automatic1111 (stable diffusion).  Just be aware going forward things may behave non-intuitively based on previous performance.   After upgrading I chose to roll back.

**Comment by LostRuins** [comment]
Good to know. I guess people may be surprised when offloading more layers results in slower performance. Not sure how to handle this.

**Comment by brknsoul** [comment]
If this is still going, iirc the last driver version before nvidia went in this direction was 531.

**Comment by pacmanincarnate** [comment]
is there any idea if pushing into shared memory is slower than running off of RAM? If I'm splitting a model between VRAM and RAM already, is there an efficiency loss in increasing my GPU layers and pushing more into shared vs just RAM?

**Comment by berkut1** [comment]
The Nvidia community manager said that they will fix that in the next driver update. https://github.com/vladmandic/automatic/discussions/1285#discussioncomment-6328116
